test/statistics.jl

```diff
@@ -1,47 +1,191 @@
+    n = length(x)
+    @testset "std unweighted" begin
+        s = std(M, x; kwargs...)
+        if sexp !== nothing
+            @test s ≈ sexp
+        end
+        @test s == std(M, x; corrected = true, kwargs...)
+        @test s ≈ √var(M, x; corrected = true, kwargs...)
+        _, s2 = mean_and_std(M, x; kwargs...)
+        @test s2 == s
+        m = mean(M, x; kwargs...)
+        @test std(M, x, m; kwargs...) == std(M, x; kwargs...)
+        @test std(M, x; corrected = false, kwargs...) == std(M, x, m; corrected = false, kwargs...)
+        @test std(M, x; corrected = false, kwargs...) ≈ std(M, x; kwargs...) * sqrt((n - 1) / n)
+        @test std(M, x, m; corrected = false, kwargs...) ≈ std(M, x, m; kwargs...) * sqrt((n - 1) / n)
+    end
+
+    @testset "std weighted" begin
+        n = length(x)
+        w1 = pweights(ones(n) / n)
+        w2 = pweights(ones(n))
+        w3 = pweights(2 * ones(n))
+
+        s = std(M, x; corrected = false, kwargs...)
+        for w in (w1, w2, w3)
+            @test std(M, x, w; kwargs...) ≈ s
+            @test std(M, x, w; corrected = false, kwargs...) ≈ s
+            @test mean_and_std(M, x, w; kwargs...)[2] == std(M, x, w; kwargs...)
+            m = mean(M, x, w; kwargs...)
+            @test std(M, x, w, m; kwargs...) == std(M, x, w; kwargs...)
+            @test std(M, x, w; corrected = true, kwargs...) == std(M, x, w, m; corrected = true, kwargs...)
+            @test std(M, x, w; corrected = true, kwargs...) ≈ std(M, x, w; kwargs...) * sqrt(n / (n - 1))
+            @test std(M, x, w, m; corrected = true, kwargs...) ≈ std(M, x, w, m; kwargs...) * sqrt(n / (n - 1))
+        end
+        @test_throws DimensionMismatch std(M, x, pweights(ones(n + 1)); kwargs...)
+    end
+end
+
+@testset "Statistics" begin
+    @testset "TestStatsSphere" begin
+        M = TestStatsSphere(2)
+
+        @testset "consistency" begin
+            p = [0.,0.,1.]
+            n=3
+            x = [ exp(M,p,2/n*[cos(α), sin(α), 0.]) for α = range(0,2*π - 2*π/n, length=n) ]
+            test_mean(M, x)
+            test_median(M, x; rng = MersenneTwister(1212), atol = 10^-12)
+            test_var(M, x)
+            test_std(M, x)
+        end
+
+        @testset "zero variance" begin
+            x = fill([0.,0.,1.], 5)
+            test_mean(M, x, [0.,0.,1.])
+            test_median(M, x, [0.,0.,1.]; rng = MersenneTwister(1212), atol = 10^-12)
+            test_var(M, x, 0.0)
+            test_std(M, x, 0.0)
+        end
+
+        @testset "two points" begin
+            x = [ [1., 0., 0.], [0., 1., 0.] ]
+            θ = π / 4
+            @test isapprox(M, mean(M, x), geodesic(M, x[1], x[2], θ))
+            test_mean(M, x, [1.0, 1.0, 0.0] / √2)
+            test_median(M, x, [1.0, 1.0, 0.0] / √2; rng = MersenneTwister(1212), atol = 10^-12)
+            test_var(M, x, θ^2 * 2)
+            test_std(M, x, θ * √2)
+        end
+    end
+
+    @testset "Euclidean fallbacks" begin
```

Comment:
I think we should also test the mutating fallbacks.

Reply:
Thanks for the suggestion. I wrote the tests and they methods they fall back to don't work with those types, so I removed them.